cohttp-server-lwt-unix/src/cohttp_server_lwt_unix.ml

```diff
@@ -16,6 +16,24 @@
 
 open Lwt.Syntax
 
+module Direct_access = struct
+  let rec write_sub (da : Lwt_io.direct_access) s ~pos ~len =
+    if len = 0 then Lwt.return_unit
+    else
+      let remaining = da.da_max - da.da_ptr in
+      if remaining = 0 then
+        let* (_ : int) = da.da_perform () in
```

Comment:
`da_perform` can raise an exception I believe? We might want to handle that via a `Lwt.catch`

Reply:
It can indeed raise Eof, BADFD, etc. There's not much we can do apart from aborting the connection though. As long as our finalizers are cleaning everything up it should be fine.

Reply:
That sounds good!